src/index.js

```diff
@@ -590,14 +591,15 @@ class DataProvider {
      * resource.data contains rows with all the fields
      * resource.fields specifies which fields to aggregate
      */
-    const aggregatedData = {};
-
     let mainResource = Object.values(resources).find(
       resource => resource.main === true
     );
+
+    const aggregatedData = {data: {}, total: mainResource.total};
```

Comment:
Same question as above

Reply:
the pagination wouldn't work otherwise

Reply:
I'm not sure what you mean. The pagination currently works just fine :)

Reply:
The point of this was to prepare to add the pagination, to avoid retrieving all the data to the client. The follow-up is here (I already posted it on slack a while back):
https://github.com/dryhten/ra-resource-aggregator/compare/master...leibowitz:fix/pagination-page?expand=1
The main issue with this branch was that the count query has the filters hardcoded. There is no way to add a filter (to return only the items we want in a join) and return the total (without the filter applied) at the same time without making a change to the underlying provider (ra-data-hasura), so it returns the same total as the number of items on the page requested (when applying a filter). This can be fixed easily, but I gave up on that

Reply:
Can you move these changes in a different PR then with the one you just linked as well? I don't recall looking over it and it probably makes more sense to discuss it there? Maybe also discuss it in an issue?

> The main issue with this branch was that the count query has the filters hardcoded. 
I assume you mean for non main tables?

Like I said, probably better to discuss it in a separate PR/issue. The changes to avoid duplicate queries to retrieve count in my opinion are the ones from `getAllRecords` and I approve of them.